### PR TITLE
Testing/login page

### PR DESCRIPTION
**Pull request by aominhlong**
## Description 
Add testing to the 'Create user' and 'Login' page

#### Where to start?
Start in the 'login-flow.cy.js' file

#### How to test?
In the project directory, you can run:

### `npm start`
Runs the app in the development mode.\
Open [http://localhost:3000](http://localhost:3000) to view it in your browser.
The page will reload when you make changes.

### `npm run cypress`
-click 'login-flow' in the E2E specs section
-The test will automatically run.

#### Screenshots (if appropriate)
image

#### Type of Change:
- [] Bug fix (non-breaking change which fixes an issue)
- [] Refactor (code is working and without bugs, formatting has been streamlined)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)

#### How Has This Been Tested?
What steps have you taken to ensure the code is effective?
- [x] Have you used Dev Tools/ ran code in the console?
- [x] Have you ran the Cypress testing suite?

#### Checklist:
- [x] My code follows the style guidelines of this project.
- [x] My commit messages are consistent, descriptive, concise and begin with a verb and capital letter.
- [x] I have performed a self-review of my own code.
- [x] I have added reviewers.

### CODE REVIEW

File: cypress/e2e/login-flow.cy.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    cy.get('.input-container > :nth-child(5)').should('have.text', 'If already a user, click Here to login')
+  })
+
+  it('should create a user and be taken to the confirmation page', () => {
+    cy.intercept('POST', 'https://pup-trainer-api.herokuapp.com/graphql', (req) => {
+      const { body } = req
+      aliasMutation(req, 'createUser')
+      if (hasOperationName(req, 'createUser')) {
+        req.alis = 'gqlcreateUserMutation'
+        req.reply((res) => {
```
Comment (Jlavarine) [inline cypress/e2e/login-flow.cy.js:46]: Great job mocking the response from our queries!

**Review by Jlavarine — APPROVED** [review]
Looks great!

File: cypress/e2e/login-flow.cy.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    cy.url().should('eq', 'http://localhost:3000/create-user')
+  })
+
+  it('should display the create-user text fields', () => {
+    cy.visit('http://localhost:3000/create-user')
+    cy.get('.input-container > :nth-child(1)').should('have.text', 'Please enter a username and email')
+    cy.get('.input-container > :nth-child(5)').should('have.text', 'If already a user, click Here to login')
+  })
+
+  it('should create a user and be taken to the confirmation page', () => {
```
Comment (DanielN88) [inline cypress/e2e/login-flow.cy.js:40]: This looks good for stubbing the data.

**Review by DanielN88 — APPROVED** [review]
Everything looks good.

File: cypress/e2e/login-flow.cy.js (hunk 1 above)
Comment (JoshMallery) [reply cypress/e2e/login-flow.cy.js:46]: solid work on getting the stubs and intercepts working!

**Review by JoshMallery — APPROVED** [review]
Thanks for getting the testing going, the login page has several calls.

**Pull request merged**